app/helpers/application_helper.rb

```diff
@@ -35,14 +36,13 @@ def current_locale
     I18n.locale
   end
 
-  def billable_entry_checkbox(entry)
+  def billable_entry_checkbox(entry, entry_type)
     if entry.billed
       "√"
     else
-      tag(:input, type: "checkbox", name: "entries_to_bill[]", class: "bill_checkbox", value: entry.id, "data-project-id" => entry.project.id)
+      tag(:input, type: "checkbox", name: "entries_to_bill[]", class: "bill_checkbox", value: "#{entry_type}-#{entry.id}", "data-project-id" => entry.project.id, "data-entry_type" => entry_type)
```

Comment:
The value is for the controller. the data entry_type is for the Js to be able to make the 'mark all' button work. If the JS can be rewritten to use the value instead, then only one would be needed. Not sure if that's possible, I'll look into it.